go/services/reports/pkg/scheduled_jobs/scheduled_job_repository.go

```diff
@@ -0,0 +1,175 @@
+package scheduledjobs
+
+import (
+	"context"
+	"fmt"
+	"github.com/IBM/fp-go/array"
+	"github.com/Magpie-Monitor/magpie-monitor/pkg/repositories"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.uber.org/zap"
+)
+
+type ScheduledJobRepositoryErrorKind string
+
+type ScheduledJob interface {
+	GetId() string
+	SetId(newId string)
+	GetStatus() string
+}
+
+const (
+	ScheduledJobNotFound      ScheduledJobRepositoryErrorKind = "SCHEDULED_JOB_NOT_FOUND"
```

Comment:
QQ regarding the naming, in OpenAiJobStatus a different naming convention was used (with __ in betwen). Is that inconsistency intentional?

Reply:
No it wasn't intentional, thanks!
I will name it the same way I did it with OpenAiJobStatus